File: frontend/src/constants/pet.ts
export type PetSpecies = 'dog' | 'cat' | 'rabbit' | 'other'
export type PetStatus = 'available' | 'pending' | 'adopted'

export const PetSpeciesMap: Record<PetSpecies, string> = {
  dog: '犬',
  cat: '猫',
  rabbit: '兔',
  other: '其他',
}

export const PetStatusMap: Record<PetStatus, { text: string; color: string }> = {
  available: { text: '可领养', color: 'green' },
  pending: { text: '申请中', color: 'gold' },
  adopted: { text: '已领养', color: 'default' },
}

export interface Pet {
  id: number
  org_id: number
  name: string
  species: PetSpecies
  breed: string
  age: number
  gender: string
  size: string
  city: string
  description: string
  personality: string
  health_status: string
  neutered: boolean
  vaccinated: boolean
  image_urls: string
  status: PetStatus
  created_at: string
}

export function parseImages(raw: string): string[] {
  try {
    const arr = JSON.parse(raw || '[]')
    return Array.isArray(arr) ? arr : []
  } catch {
    return []
  }
}

File: backend/internal/model/pet.go
package model

import "time"

// Pet is an adoptable animal published by an organization.
type Pet struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	OrgID        uint      `gorm:"index;not null" json:"org_id"`
	Name         string    `gorm:"size:64;not null" json:"name"`
	Species      string    `gorm:"size:16;index;not null" json:"species"`
	Breed        string    `gorm:"size:64" json:"breed"`
	Age          int       `json:"age"`
	Gender       string    `gorm:"size:8" json:"gender"`
	Size         string    `gorm:"size:16" json:"size"`
	City         string    `gorm:"size:64;index" json:"city"`
	Description  string    `gorm:"type:text" json:"description"`
	Personality  string    `gorm:"size:255" json:"personality"`
	HealthStatus string    `gorm:"size:128" json:"health_status"`
	Neutered     bool      `json:"neutered"`
	Vaccinated   bool      `json:"vaccinated"`
	ImageURLs    string    `gorm:"type:json" json:"image_urls"`
	Status       string    `gorm:"size:16;default:available;index" json:"status"`
	CreatedAt    time.Time `json:"created_at"`
}
